common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java

```diff
@@ -582,9 +582,27 @@ void addSpan(SpanRecord span) {
     }
   }
 
+  /**
+   * Thread-specific random number generators. Each is seeded with the thread
+   * ID, so the sequence of pseudo-random numbers are unique between threads.
+   *
+   * See http://stackoverflow.com/questions/2546078/java-random-long-number-in-0-x-n-range
+   */
+  private static ThreadLocal<Random> random = new ThreadLocal<Random>() {
+      @Override
+      protected Random initialValue() {
+        // It'd be nice to get the process ID into the mix, but there's no clear
+        // cross-platform, Java 6-compatible way to determine that
+        return new Random(
+          System.currentTimeMillis() *
```

Comment:
seems like a great time to use `nanoTime`, though maybe that's not available in Android??